### DISCUSSION TITLE
Generally preferred python version

### DISCUSSION THREAD

**Discussion by buckleyc** [ssantoshp/Empyrial#34] [Q&A]
I was looking at Empyrial/empyrial.py, with the intention of cleaning up the code. In general, I mean items such as proper and consistent indenting, cleaner comments, consistent spacing, etc. 
To that end, there are some general issues surrounding some conditionals that I would want to clean up. But I want to be sure that the group would be comfortable with this.  I generally use python 3.9 (currently running 3.9.5 & 3.9.6). Any code modification would be usable from 3.7 and later.  
I was wondering if anyone is trying to use or maintain compatibility with python versions earlier than 3.7.

**Comment by santoshlite** [comment]
Hi @buckleyc,

Thank you for contributing 😸 . I think this should be fine for most users here. So far, no user has raised an issue related to a python version problem. 3.7 should be fine!

**Reply by buckleyc** [reply]
Okay; with that in mind, I did a large rewrite to empyrial.py today, and have committed these.  Please note, that there should be no functional impact, and there were actually a number of bug fixes. You should see the Pull Request now.  Cheers.

**Reply by santoshlite** [reply]
Ok, I'll check that and merge it, thank you so much for this contribution!

**Comment by santoshlite** [comment]
Hi @buckleyc, I changed the version but it gave me an error as you can see here: https://colab.research.google.com/drive/1NrCZmIZP9oTUlSGufacQrQWxuCzGxJ9M?usp=sharing

**Reply by buckleyc** [reply]
I am seeing something similar. Unsure of which change that would have caused this. Looking into this now.

**Reply by buckleyc** [reply]
I may have found the issue. Working on it.

**Reply by buckleyc** [reply]
Okay - this should be fixed. The major issue was that I had omitted the 'not' in the conditional 'if self.rebalance is not None', which made everything after it fail.  I still think this needs to be handled better. Particularly the conditional 'if self.rebalance is not None:' (now) starting at line 90; so much use of the word 'rebalance'.  I also cleaned up the handling for when the optimizer is called from the dict. 
Please give it a try so we can confirm it is working as you expect.

**Reply by santoshlite** [reply]
That worked! Thankss
